### DISCUSSION TITLE
Metamask V2 Integration : Internal Error : {"code": -32603, "message": "Internal error"}

### DISCUSSION THREAD

**Discussion by jobinabraham** [WalletConnect/walletconnect-monorepo#2784] [Developer Support]
Hey Community...Trying to migrate v1 react-native dapp implementation to v2 in my react-native application. When trying to migrate "sendTransation" call from V1 to "provider.request" in V2, getting JSON-RPC error from metamask. The v1 version is working correctly so I am assuming its not an issue with metamask. No params have changed just the function. Please help me out.
`import {
WalletConnectModal,
useWalletConnectModal,
} from '@walletconnect/modal-react-native';
const connector = useWalletConnectModal();

//----------V1 Version start--------------------//
connector.sendTransaction({
data: encodedData,
from: account,
to: tokenAddress,
nonce,
maxFee: 1573479400382,
maxPriorityFee: 115446958670,
})
//----------V1 Version End--------------------//
//----------V2 Version Start------------------//
connector?.provider
?.request({
method: 'eth_sendTransaction',
params: [{
data: encodedData,
from: account,
to: tokenAddress,
nonce,
maxFee: 1573479400382,
maxPriorityFee: 115446958670,
},
],
})
//----------V2 Version End--------------------//

Error from Metamask: {"code": -32603, "message": "Internal error"}`

It seems somehow the maxFee and MaxFeePerUser params are not being read properly or passed through to metamask-mobile. Not sure how to fix this.

**Comment by chadyj** [comment]
Related: https://github.com/WalletConnect/walletconnect-monorepo/issues/2825
